crossing-repair.lic

```diff
@@ -2,51 +2,199 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#crossing-repair
 =end
 
-custom_require.call(%w[common common-money common-travel drinfomon equipmanager])
+custom_require.call(%w[common common-money common-travel equipmanager])
 
 class CrossingRepair
-  include DRC
-  include DRCM
-  include DRCT
 
   def initialize
-    settings = setup
+    arg_definitions = [
+      [
+        { name: 'town', regex: $HOMETOWN_REGEX, optional: true, description: 'Town to repair in' },
+        { name: 'force', regex: /force/i, optional: true, description: 'Force repair gear, ignoring any wait timers' },
+        { name: 'debug', regex: /debug/i, optional: true, description: 'Enable debug mode' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    $debug_mode_crossing_repair = args.debug || UserVars.crossing_repair_debug
+    town = DRC.get_town_name(args.town)
+    settings = get_settings
```

Comment:
It's _super_ minor, but we could delay getting settings until after we validate `town` (i.e. the potentially early exit of the script). It doesn't matter right now since get settings is so fast. Just a general thing to think about.

Reply:
Alternatively, what do you think about moving the town check here and the 'should?' check on line 50 into `repair_prep`? That consolidates the early exit checks in `repair_prep`...

If we did that, I'd also suggest pulling the call to `repair_prep` out of `repair_handler`, so line 67 becomes something like:

```
    # Calls the repair prep and validation method. Makes sure we're ready and able to repair.
    # Grabs a hash of the items to repair at leather and metal, respectively.
    repair_info = repair_prep(town, settings)

    # Start performance script for some sweet TDPs.
    start_script('performance', ['noclean']) unless settings.instrument

    # Call our main repair handler function.
    repair_handler(repair_info, town, settings)
```